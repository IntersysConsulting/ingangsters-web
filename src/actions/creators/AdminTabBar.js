import * as types from "../types/AdminTabBar";

export function setActiveAdminBarButton(newButton) {
  return { type: types.SET_ACTIVE_ADMINBAR_BUTTON, newButton };
}
