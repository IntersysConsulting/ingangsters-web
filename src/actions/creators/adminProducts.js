import * as types from "../types/adminProducts";
import { API } from "../../config";
import axios from "axios";

export const updateAdminProducts = newProductList => ({
  type: types.UPDATE_PRODUCTS,
  newProductList
});

export const startFetchProducts = () => ({
  type: types.FETCHING_PRODUCTS
});

export const endFetchProducts = (totalItems, itemsPerPage, currentPage) => ({
  type: types.FINISHED_FETCHING_PRODUCTS,
  totalItems,
  itemsPerPage,
  currentPage
});

export const updatePaginator = (totalItems, itemsPerPage, currentPage) => ({
  type: types.UPADTE_PAGINATOR,
  totalItems,
  itemsPerPage,
  currentPage
});

export const fetchProducts = pageRequested => async dispatch => {
  const numberOfProducts = window.innerWidth <= 550 ? 1 : 1;
  const config = {
    headers: {
      "Content-Type": "application/json"
    }
  };
  const endpoint = `${API}/products/${numberOfProducts}/${pageRequested}`;
  try {
    dispatch(startFetchProducts());
    const result = await axios.get(endpoint, config);
    const { products, total_products } = result.data.data;
    dispatch(updateAdminProducts(products));
    dispatch(endFetchProducts(total_products, numberOfProducts, pageRequested));
  } catch (err) {
    console.log(err);
  }
};
