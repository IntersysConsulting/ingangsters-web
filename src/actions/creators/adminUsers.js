import * as types from "../types/adminUsers";
import { API } from "../../config";
import axios from "axios";

export const updateAdminUsers = adminUserList => ({
  type: types.UPDATE_ADMIN_USER,
  adminUserList
});

export const startFetchAdminUsers = () => ({
  type: types.FETCHING_ADMIN_USERS
});

export const endFetchAdminUsers = (totalUsers, usersPerPage, currentPage) => ({
  type: types.FINISHED_FETCHING_ADMIN_USERS,
  totalUsers,
  usersPerPage,
  currentPage
});

export const updatePaginator = (totalUsers, usersPerPage, currentPage) => ({
  type: types.UPADTE_PAGINATOR,
  totalUsers,
  usersPerPage,
  currentPage
});

export const fetchAdminUsers = pageRequested => async dispatch => {
  const numberOfUsers = window.innerWidth <= 550 ? 3 : 3;
  const AuthStr = `Bearer ${localStorage.getItem("token")}`;

  const endpoint = `${API}/admin/${numberOfUsers}/${pageRequested}`;
  try {
    dispatch(startFetchAdminUsers());
    const result = await axios.get(endpoint, {
      headers: { "Content-Type": "application/json", Authorization: AuthStr }
    });

    const adminUsers = result.data.data.admin_users;
    const total_admin_users = result.data.data.total_users;

    console.log(total_admin_users);
    dispatch(updateAdminUsers(adminUsers));
    dispatch(
      endFetchAdminUsers(total_admin_users, numberOfUsers, pageRequested)
    );
  } catch (err) {
    console.log(err);
  }
};

export const getAdminUsers = () => async dispatch => {
  const AuthStr = `Bearer ${localStorage.getItem("token")}`;
  try {
    const res = await axios.get(`${API}/admin/users`, {
      headers: { "Content-Type": "application/json", Authorization: AuthStr }
    });
    dispatch({
      type: types.GET_ADMIN_USER,
      payload: res.data
    });
  } catch (err) {
    console.log(err);
    dispatch({
      type: types.ADMIN_USERS_ERROR,
      payload: err
    });
  }
};
