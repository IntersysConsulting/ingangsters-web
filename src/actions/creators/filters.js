import * as types from "../types/filters";
import store from "../../store";

export const enableOrderByPriceLTH = () => ({
  type: types.ORDER_BY_ENABLE_PRICE_LTH
});

export const enableOrderByPriceHTL = () => ({
  type: types.ORDER_BY_ENABLE_PRICE_HTL
});

export const enableOrderByNameAZ = () => ({
  type: types.ORDER_BY_ENABLE_NAME_AZ
});

export const enableOrderByNameZA = () => ({
  type: types.ORDER_BY_ENABLE_NAME_ZA
});

export const enableOrderByStockLTH = () => ({
  type: types.ORDER_BY_ENABLE_STOCK_LTH
});

export const enableOrderByStockHTL = () => ({
  type: types.ORDER_BY_ENABLE_STOCK_HTL
});

export const disableOrderBy = () => ({
  type: types.ORDER_BY_CLEAR_ALL
});

export const setShippableTrueFilter = value => ({
  type: types.FILTERS_SET_SHIPPABLE_TRUE,
  enable: value
});

export const setShippableFalseFilter = value => ({
  type: types.FILTERS_SET_SHIPPABLE_FALSE,
  enable: value
});

export const setAvailableTrueFilter = value => ({
  type: types.FILTERS_SET_AVAILABLE_TRUE,
  enable: value
});

export const setAvailableFalseFilter = value => ({
  type: types.FILTERS_SET_AVAILABLE_FALSE,
  enable: value
});

export const setPriceRangeFilter = (val, min, max) => ({
  type: types.FILTERS_SET_PRICE_RANGE,
  data: {
    enabled: val,
    min: min,
    max: max
  }
});

export const clearFilters = () => ({
  type: types.FILTERS_CLEAR_FILTERS
});

export const setSearch = search => ({
  type: types.SEARCH_PRODUCTS,
  data: search
});

export function getFilterParams() {
  const filtersConfig = store.getState().filters.filtersEnabled;
  const orderConfig = store.getState().filters.orderCriteria;
  const adminProductSearch = store.getState().filters.search;
  const params = {
    sortPrice: "NONE",
    sortName: "NONE",
    sortStock: "NONE",
    shippable: "NONE",
    available: "NONE",
    priceRange: "NONE",
    search: "NONE"
  };

  if (adminProductSearch !== "") params.search = adminProductSearch;
  else delete params.search;

  if (orderConfig.priceLTH) params.sortPrice = "ASCENDING";
  else if (orderConfig.priceHTL) params.sortPrice = "DESCENDING";

  if (orderConfig.nameAZ) params.sortName = "ASCENDING";
  else if (orderConfig.nameZA) params.sortName = "DESCENDING";

  if (orderConfig.stockLTH) params.sortStock = "ASCENDING";
  else if (orderConfig.stockHTL) params.sortStock = "DESCENDING";

  if (filtersConfig.availableTrue && filtersConfig.availableFalse)
    delete params.available;
  else if (filtersConfig.availableTrue) params.available = "TRUE";
  else if (filtersConfig.availableFalse) params.available = "FALSE";

  if (filtersConfig.shippableTrue && filtersConfig.shippableFalse)
    delete params.shippable;
  else if (filtersConfig.shippableTrue) params.shippable = "TRUE";
  else if (filtersConfig.shippableFalse) params.shippable = "FALSE";

  if (filtersConfig.priceRange.enabled) {
    params.min = filtersConfig.priceRange.min;
    params.max = filtersConfig.priceRange.max;
  }
  return params;
}
