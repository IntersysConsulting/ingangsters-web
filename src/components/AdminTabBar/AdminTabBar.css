.AdminTabBar {
  text-align: center;
  align-self: center;
  font-family: "Roboto", sans-serif;
  font-size: 14px;
}
.AdminTabBar button {
  background-color: var(--green);
  border-color: var(--green);
  width: 100%;
  height: 100%;
}
.AdminTabBar button:hover {
  background-color: var(--green);
  border-color: var(--gren);
  box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24),
    0 17px 50px 0 rgba(0, 0, 0, 0.19);
}
.AdminTabBar button:focus {
  background-color: var(--green);
  border-color: var(--gren);
  box-shadow: 0 4px 8px 0 #000000, 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}
