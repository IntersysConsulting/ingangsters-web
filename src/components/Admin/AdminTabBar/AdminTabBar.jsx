import React from "react";
import { Link } from "react-router-dom";
import ButtonGroup from "react-bootstrap/ButtonGroup";
import Button from "react-bootstrap/Button";
import "./AdminTabBar.css";
import "../../../css/colors.css";
import { connect } from "react-redux";
import { setActiveAdminBarButton } from "../../../actions/creators/AdminTabBar";

const AdminTabBar = ({ handleClick }) => {
  return (
    <div className="AdminTabBar">
      <ButtonGroup className="buttonGroupAdmin" size="lg">
        <Link id="firstTabBarButtonLink" to="/admin/dashboard">
          <Button
            className="tabBarFirstButton"
            onClick={handleClick}
            value="Users"
          >
            Users
          </Button>
        </Link>

        <Link to="/admin/dashboard">
          <Button onClick={handleClick} value="Products">
            Products
          </Button>
        </Link>

        <Link to="/admin/dashboard">
          <Button
            className="tabBarLastButton"
            onClick={handleClick}
            value="Orders"
          >
            Orders
          </Button>
        </Link>
      </ButtonGroup>
      <br />
      <br />
    </div>
  );
};

function mapStateToProps(state) {
  return {
    active: state.adminTabBar.activeButton
  };
}

function mapDispatchToProps(dispatch) {
  return {
    handleClick(e) {
      dispatch(setActiveAdminBarButton(e.target.value));
      console.log("State Changed");
    }
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminTabBar);
