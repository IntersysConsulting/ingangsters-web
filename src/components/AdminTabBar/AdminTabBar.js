import React, { Component } from "react";
import "./AdminTabBar.css";
import "../../css/colors.css";

class AdminTabBar extends Component {
  render() {
    return <div></div>;
  }
}
export default AdminTabBar;
