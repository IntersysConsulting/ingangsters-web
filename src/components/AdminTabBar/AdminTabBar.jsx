import React, { Component } from "react";
import ButtonGroup from "react-bootstrap/ButtonGroup";
import Button from "react-bootstrap/Button";
import "./AdminTabBar.css";
import "../../css/colors.css";
import { connect } from "react-redux";
import { setActiveAdminBarButton } from "../../actions/creators/AdminTabBar";

const AdminTabBar = ({ active, handleClick }) => {
  console.log(active);
  return (
    <div className="AdminTabBar">
      <ButtonGroup size="lg">
        <Button onClick={handleClick} value="Users">
          Users
        </Button>
        <Button onClick={handleClick} value="Products">
          Products
        </Button>
        <Button onClick={handleClick} value="Orders">
          Orders
        </Button>
      </ButtonGroup>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    active: state.shadow.active
  };
}

function mapDispatchToProps(dispatch) {
  return {
    handleClick(e) {
      console.log("Click");
      dispatch(setActiveAdminBarButton(e.target.value));
    }
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminTabBar);
