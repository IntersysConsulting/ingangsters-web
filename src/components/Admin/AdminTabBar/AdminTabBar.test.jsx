import React from "react";
import { mount } from "enzyme";
import AdminTabBar from "./AdminTabBar";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { setActiveAdminBarButton } from "../../../actions/creators/AdminTabBar";
import { fetchProducts } from "../../../actions/creators/adminProducts";
import { fetchAdminUsers } from "../../../actions/creators/adminUsers";
import { FETCHING_ADMIN_USERS } from "../../../actions/types/adminUsers";
import { FETCHING_PRODUCTS } from "../../../actions/types/adminProducts";
import { CLEAR_SEARCH } from "../../../actions/types/filters";
import * as types from "../../../actions/types/AdminTabBar";
import reducer from "../../../reducers/AdminTabBar";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";

describe("AdminTabBar", () => {
  const initialState = {
    adminTabBar: {
      activeButton: "Users"
    }
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const actions = [setActiveAdminBarButton, fetchProducts, fetchAdminUsers];
  const store = mockStore(initialState, actions);

  it("renders correctly", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <AdminTabBar store={store} dispatch={null} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  const wrapper = mount(
    <MemoryRouter>
      <AdminTabBar store={store} dispatch={null} />
    </MemoryRouter>
  );

  it("Buttons redirect to admin/dashboard", () => {
    let linkElement = wrapper.find("#adminTabBarUsersDiv");
    expect(linkElement.find("Link").prop("to")).toBe("/admin/dashboard");

    linkElement = wrapper.find("#adminTabBarProductsDiv");
    expect(linkElement.find("Link").prop("to")).toBe("/admin/dashboard");

    linkElement = wrapper.find("#adminTabBarOrdersDiv");
    expect(linkElement.find("Link").prop("to")).toBe("/admin/dashboard");
  });

  it("setActiveAdminBarButton Action Creator", () => {
    const newButton = "Users";
    const expectedAction = {
      type: types.SET_ACTIVE_ADMINBAR_BUTTON,
      newButton
    };
    expect(setActiveAdminBarButton(newButton)).toEqual(expectedAction);
  });

  it("should handle SET_ACTIVE_ADMINBAR_BUTTON", () => {
    expect(
      reducer([], {
        type: types.SET_ACTIVE_ADMINBAR_BUTTON,
        newButton: "Users"
      })
    ).toEqual({
      activeButton: "Users"
    });

    expect(
      reducer([], {
        type: types.SET_ACTIVE_ADMINBAR_BUTTON,
        newButton: "Products"
      })
    ).toEqual({
      activeButton: "Products"
    });

    expect(
      reducer([], {
        type: types.SET_ACTIVE_ADMINBAR_BUTTON,
        newButton: "Orders"
      })
    ).toEqual({
      activeButton: "Orders"
    });
  });

  it("handles Click", () => {
    let event = { target: { value: "Users" } };
    wrapper
      .find("#adminTabBarUsersButton")
      .at(0)
      .simulate("click", event);
    expect(store.getActions()).toEqual([
      { newButton: "Users", type: types.SET_ACTIVE_ADMINBAR_BUTTON },
      { type: FETCHING_ADMIN_USERS }
    ]);

    event = { target: { value: "Products" } };
    store.clearActions();
    wrapper
      .find("#adminTabBarProductsButton")
      .at(0)
      .simulate("click", event);
    expect(store.getActions()).toEqual([
      { newButton: "Products", type: types.SET_ACTIVE_ADMINBAR_BUTTON },
      { type: CLEAR_SEARCH },
      { type: FETCHING_PRODUCTS }
    ]);

    event = { target: { value: "Orders" } };
    store.clearActions();
    wrapper
      .find("#adminTabBarOrdersButton")
      .at(0)
      .simulate("click", event);
    expect(store.getActions()).toEqual([
      { newButton: "Orders", type: types.SET_ACTIVE_ADMINBAR_BUTTON }
    ]);
  });

  it("Has 3 Buttons with Links", () => {
    expect(wrapper.find("Link").length).toEqual(3);
    expect(wrapper.find("Button").length).toEqual(3);
  });
});
