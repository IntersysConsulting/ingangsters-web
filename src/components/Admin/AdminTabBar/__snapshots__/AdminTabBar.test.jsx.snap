// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminTabBar renders correctly 1`] = `
<div
  className="AdminTabBar"
>
  <div
    className="buttonGroupAdmin btn-group btn-group-lg"
    role="group"
  >
    <div
      id="adminTabBarUsersDiv"
    >
      <a
        href="/admin/dashboard"
        onClick={[Function]}
      >
        <button
          className="tabBarFirstButton btn btn-primary"
          disabled={false}
          id="adminTabBarUsersButton"
          onClick={[Function]}
          type="button"
          value="Users"
        >
          Users
        </button>
      </a>
    </div>
    <div
      id="adminTabBarProductsDiv"
    >
      <a
        href="/admin/dashboard"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          disabled={false}
          id="adminTabBarProductsButton"
          onClick={[Function]}
          type="button"
          value="Products"
        >
          Products
        </button>
      </a>
    </div>
    <div
      id="adminTabBarOrdersDiv"
    >
      <a
        href="/admin/dashboard"
        onClick={[Function]}
      >
        <button
          className="tabBarLastButton btn btn-primary"
          disabled={false}
          id="adminTabBarOrdersButton"
          onClick={[Function]}
          type="button"
          value="Orders"
        >
          Orders
        </button>
      </a>
    </div>
  </div>
  <br />
  <br />
</div>
`;
